library/core/src/cell.rs

```diff
@@ -1893,7 +1893,8 @@ impl<T: ?Sized + fmt::Display> fmt::Display for RefMut<'_, T> {
 /// on an _exclusive_ `UnsafeCell<T>`. Even though `T` and `UnsafeCell<T>` have the
 /// same memory layout, the following is not allowed and undefined behavior:
 ///
-/// ```rust,no_run
+#[cfg_attr(bootstrap, doc = "```rust,no_run")]
+#[cfg_attr(not(bootstrap), doc = "```rust,compile_fail")]
```

Comment:
This is an interesting case. It's a UB example but it's also something that's fine in both SB and TB as they don't care about `get` at all, so it seems likely to be fine in the future. Avoiding to lint here is quite easy by just checking whether the pointee has interior mutability with `!pointee.is_freeze()`, but I'm afraid that this would cause the lint to miss more subtle cases where the type has interior mutability somewhere deeply nested.
@RalfJung what's your opinion on this here? Should the compiler emit a deny-by-default lint on this code below?

Reply:
I am totally fine with linting against this, calling `get` is a lot more clear and the docs for UnsafeCell explicitly say this is UB. It might be "just" library UB instead of language UB, but that's still good enough to justify a lint IMO. We can always revisit this once we have an official aliasing model.